agent/engine/docker_image_manager.go

```diff
@@ -72,6 +74,19 @@ func NewImageManager(cfg *config.Config, client DockerClient, state dockerstate.
 	}
 }
 
+// WhitelistImage add the image into whitelists that won't be cleaned up by agent
```

Comment:
Not sure I really like the term whitelist here.  I think it's clearer if we say `PreserveImage` or `MarkImagePersistent`.